Sources/MultipartKit/MultipartParserError.swift

```diff
@@ -1,7 +1,25 @@
 /// Technical parsing error, such as malformed data or invalid characters.
 /// This is mainly used by ``MultipartParser``.
-package enum MultipartParserError: Swift.Error, Equatable {
-    case invalidBoundary
-    case invalidHeader(reason: String)
-    case invalidBody(reason: String)
+public struct MultipartParserError: Swift.Error, Equatable {
+
+    enum Base: Equatable {
+        case invalidBoundary
+        case invalidHeader(reason: String)
+        case invalidBody(reason: String)
+    }
+
+    let base: Base
+
+    private init(base: Base) {
```

Comment:
nit pick: use `_` like the other code:
```suggestion
    private init(_ base: Base) {
```